Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/util/SecurityUtils.java

```diff
@@ -0,0 +1,245 @@
+import javax.crypto.spec.SecretKeySpec;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.security.*;
+import java.security.cert.Certificate;
+import java.util.Arrays;
+import java.util.Base64;
+
+public final class SecurityUtils {
+
+    private static final Logger logger = LoggerFactory.getLogger(SecurityUtils.class);
+
+    private static final String KEYSTORE_FILE = "apSecurityTS.jks";
+
+    private static SecretKeySpec secretKey;
+    private static byte[] key;
+
+    private static char[] ksPassword;
+
+    private static final String SIGNATURE_ALGORITHM = "SHA256withRSA";
+
+    private static final String KS_AND_KEY_PASSWORD_ENV_KEY = "KS_PASSWD";
+    private static final String SECURITY_RESOURCES_DIR_KEY = "SECURITY_RESOURCES_DIR";
+
+    private static final String SYMMETRIC_CIPHER_FULL_ALG_SPEC = "AES/ECB/PKCS5Padding";
+    private static final String SYMMETRIC_CIPHER_BASE_ALG = "AES";
+
+    private static final String DIGEST_ALGORITHM = "SHA1";
+    private static final String CHARSET_ENCODING = "UTF-8";
+
+    private static String secRessKeystoreFilePathStr = null;
+
+    static {
+        String pwdKSStr = System.getenv(KS_AND_KEY_PASSWORD_ENV_KEY);
+
+        if (! StringUtils.hasText(pwdKSStr)) {
+            final String errMsg = "keystore password could not be attained from environment";
+
+            logger.info("\n\n{}\n", errMsg);
+            throw new IllegalStateException(errMsg);
+        }
+
+        String resourcesDirKey = System.getenv(SECURITY_RESOURCES_DIR_KEY);
+
+        if (resourcesDirKey == null) {
+            throw new IllegalStateException("Can not proceed as security resources directory is not defined");
+        }
+
+        if (! resourcesDirKey.endsWith("/")) {
+            resourcesDirKey = resourcesDirKey + "/";
+        }
+
+        secRessKeystoreFilePathStr = resourcesDirKey + KEYSTORE_FILE;
+
+        SecurityUtils.ksPassword = pwdKSStr.toCharArray();
+        // For safety/clear
+        pwdKSStr = null;
+    }
+
+    private SecurityUtils() {
+        // Intentionally private constructor
+    }
+
+    public static final PublicKey getPublicKey(final String keyAlias) throws Exception {
+        try (
+                final InputStream inputStream = new FileInputStream(secRessKeystoreFilePathStr);
+        ) {
+            final KeyStore keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keystore.load(inputStream, SecurityUtils.ksPassword);
+
+            final Key key = keystore.getKey(keyAlias, SecurityUtils.ksPassword);
+
+            if (key instanceof PrivateKey) {
+                // Get certificate of public key
+                final Certificate cert = keystore.getCertificate(keyAlias);
+                final PublicKey publicKey = cert.getPublicKey();
+
+                return publicKey;
+            } else {
+                return null;
```

Comment:
Maybe add some `/** @return ... */` documentation mentioning that this method returns `null` if the key is absent.  Rather than throwing an exception, etc.

Reply:
Updated security functions with JavaDoc on exception case, and returning null path [DONE].